### PR TITLE
Dispatch back to IO thread for non blocking requests

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

**Comment by geoand** [comment]
I was thinking of the same thing actually.

My only question is, does this actually get you back to the same Event Loop thread that security was one?

**Comment by stuartwdouglas** [comment]
It gets you back to the requests event loop thread, security is run on a worker thread. Because this uses the context from the connection it is guaranteed to be the same thread.

**Comment by geoand** [comment]
Right, I meant the Security filter that could lead to the dispatch to the worker thread.

Sounds good

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
We would likely need to do something similar in Reactive Routes, but I assume that can wait since we are deprecating them

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/runtime/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/FlowToPublisherHandler.kt [hunk 1]
```diff
@@ -27,7 +28,9 @@ class FlowToPublisherHandler : ServerRestHandler {
                 // ensure the proper CL is not lost in dev-mode
                 Thread.currentThread().contextClassLoader = originalTCCL
                 requestContext.result = result.asMulti()
-                requestContext.resume()
+                //run in a direct invocation executor to run the rest of the invocation in the co-route scope
+                //feels a bit fragile, but let's see how it goes
+                requestContext.resume(Executor { it.run() })
```
Comment (stuartwdouglas) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/runtime/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/FlowToPublisherHandler.kt:33]: @geoand I needed to do this to make the kotlin tests work. Not sure if I like it much but I think it is probably ok for now.

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin/runtime/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/FlowToPublisherHandler.kt:33]: I think we can live with it :)

**Pull request merged**